src/components/Dataset/DatasetReady.js

```diff
@@ -49,30 +46,34 @@ export const DatasetReady = ({ dataset }) => {
                 }}
                 fill
               >
-                <CheckBox
-                  label={
-                    <Text>
-                      I agree to the{' '}
-                      <Anchor
-                        href={links.terms}
-                        label="Terms of Use"
-                        target="_blank"
-                        rel="noopener noreferrer"
-                      />
-                    </Text>
-                  }
-                  onClick={handleAgreeToTerms}
-                />
+                {!hasToken && (
+                  <CheckBox
+                    label={
+                      <Text>
+                        I agree to the{' '}
+                        <Anchor
+                          href={links.terms}
+                          label="Terms of Use"
+                          target="_blank"
+                          rel="noopener noreferrer"
+                        />
+                      </Text>
+                    }
+                    onClick={() => setTermsOfUse(!termsOfUse)}
```

Comment:
@davidsmejia Typically a "terms of ..." checkbox in a form is toggleable and we also have form validation. So let's keep this implementation as is for the time being to provide users an option. And since this is most likely a decision about the UX flow 🎨, let's ask Deepa at a later time if she would like to apply your suggestion to UI so that the terms of use checkbox will be no longer toggleble once a user checks it.